engine/src/ast/function_expr.rs

```diff
@@ -55,25 +90,37 @@ impl<'i, 's> LexWith<'i, &'s Scheme> for FunctionCallArgExpr<'s> {
         // This will provide better error reporting in most cases
         let mut chars = input.chars();
         if let Some(c) = chars.next() {
-            // check up to 3 next chars because third char of an hexa-string is either ':'
-            // or '-'
-            let c2 = chars.next();
-            let c3 = chars.next();
-            if c == '"' {
+            // lookahead the next three chars as the third char of a hexa-string
+            // is a either ':' or '-'
+            let lookahead = (c, chars.next(), chars.next());
+
+            if lookahead.0 == '"' {
                 return RhsValue::lex_with(input, Type::Bytes)
                     .map(|(literal, input)| (FunctionCallArgExpr::Literal(literal), input));
-            } else if c_is_field!(c)
-                || (c2.is_some() && c_is_field!(c2.unwrap()))
-                || (c3.is_some() && c_is_field!(c3.unwrap()))
-            {
-                return IndexExpr::lex_with(input, scheme)
-                    .map(|(lhs, input)| (FunctionCallArgExpr::IndexExpr(lhs), input));
+            }
+            if lookahead.0 == '(' {
+                return SimpleExpr::lex_with(input, scheme)
+                    .map(|(lhs, input)| (FunctionCallArgExpr::SimpleExpr(lhs), input));
+            }
+            if let Some(c2) = lookahead.1 {
+                if let Some(c3) = lookahead.2 {
+                    if is_field_char!(lookahead.0) && is_field_char!(c2) && is_field_char!(c3) {
```

Comment:
You have changed the logic here, do you have an example that exhibit the behavior you fixed?